dm/dm/master/server.go

```diff
@@ -405,6 +411,45 @@ func subtaskCfgPointersToInstances(stCfgPointers ...*config.SubTaskConfig) []con
 	return stCfgs
 }
 
+func (s *Server) initClusterID(ctx context.Context) error {
+	log.L().Info("init cluster id begin")
+	ctx1, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+
+	resp, err := s.etcdClient.Get(ctx1, common.ClusterIDKey)
+	if err != nil {
+		return err
+	}
+
+	// New cluster, generate a cluster id and backfill it to etcd
+	if len(resp.Kvs) == 0 {
+		ts := uint64(time.Now().Unix())
+		clusterID := (ts << 32) + uint64(rand.Uint32())
+		clusterIDBytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(clusterIDBytes, clusterID)
+		_, err = s.etcdClient.Put(ctx1, common.ClusterIDKey, string(clusterIDBytes))
+		if err != nil {
+			return err
+		}
+		atomic.StoreUint64(&s.clusterID, clusterID)
+		log.L().Info("generate and init cluster id success")
```

Comment:
log the new generated cluster id ?

Reply:
> log the new generated cluster id ?

Users won't use the cluster id now, Clinic system is the only user through http api.